test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetVerifyTests.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Common;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection("Dotnet Integration Tests")]
+    public class DotnetVerifyTests
+    {
+        private readonly string _noTimestamperWarningCode = NuGetLogCode.NU3027.ToString();
+        private readonly string _primarySignatureInvalidErrorCode = NuGetLogCode.NU3018.ToString();
+        private readonly string _noMatchingCertErrorCode = NuGetLogCode.NU3034.ToString();
+        private readonly string _notSignedErrorCode = NuGetLogCode.NU3004.ToString();
+
+        private MsbuildIntegrationTestFixture _msbuildFixture;
+
+        public DotnetVerifyTests(MsbuildIntegrationTestFixture fixture)
+        {
+            _msbuildFixture = fixture;
+        }
+
+        [CIOnlyFact]
+        public async Task Verify_UnSignedPackage_Fails()
+        {
+            using (var packageDir = TestDirectory.Create())
+            {
+                var packageId = "Unsigned.PackageX";
+                var packageVersion = "1.0.0";
+                var packageFile = await TestPackagesCore.GetRuntimePackageAsync(packageDir, packageId, packageVersion);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    packageDir,
+                    $"nuget verify {packageFile.FullName}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeFalse(because: result.AllOutput);
+                result.Output.Should().Contain(_notSignedErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--all")]
+        [InlineData("")]
+        public void Verify_AuthorSignedAndTimestampedPackage_Succceeds(string optionAll)
+        {
+            // Arrange
+            using (var testDirectory = TestDirectory.Create())
+            {
+                string packageX = "TestPackage.AuthorSigned.1.0.0.nupkg";
+                CopyPackageFromResources(packageX, testDirectory);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    testDirectory,
+                    $"nuget verify {Path.Combine(testDirectory, packageX)} {optionAll}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeTrue(because: result.AllOutput);
+                result.Output.Should().NotContain(_noTimestamperWarningCode);
+                result.Output.Should().NotContain(_primarySignatureInvalidErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--certificate-fingerprint 775AAB607AA76028A7CC7A873A9513FF0C3B40DF09B7B83D21689A3675B34D9A")]
+        [InlineData("--certificate-fingerprint ABC --certificate-fingerprint DEF")]
```

Comment:
According to [spec](https://github.com/NuGet/Home/blob/25cd8b91d601f133f6c442d9295d5a1909dc69a1/designs/DotNetSignCommands.md#dotnet-nuget-verify), the format should be` --certificate-fingerprint ABC;DEF`, right?
BTW, how about adding a test specifying multiple cert fingerprints and put the right one in the last?

Reply:
Spec says `This option may be supplied multiple times to provide multiple fingerprints. ` In general all the `dotnet` commands which have an option that can accept multiple values have the same behavior. For e.g. `ListPackageCommand` has an option [`--framework`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommand.cs#L41-L44) that can accept multiple values. This option was specified multiple times to provide multiple frameworks to the command in the [tests](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs#L132).

>BTW, how about adding a test specifying multiple cert fingerprints and put the right one in the last?

Sure. I will modify an [existing test case](https://github.com/NuGet/NuGet.Client/blob/dev-kartheep-ms-dotnet-nuget-verify/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetVerifyTests.cs#L94) to address this comment.

Reply:
I saw it from [nuget verify command spec](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-verify#options-for-verify--signatures): 
`Specifies one or more SHA-256 certificate fingerprints of certificates(s) which signed packages must be signed with. A certificate SHA-256 fingerprint is a SHA-256 hash of the certificate. Multiple inputs should be semicolon separated.`
Shall we keep the same behavior for dotnet command?

Reply:
@zkat - since `dotnet` commands follow POSIX style, we should not modify the current behavior of accepting multiple values for `--certificate-fingerprint` option in `dotnet nuget verify` command. correct?

Reply:
It's a shame that semicolon was chosen as the separator, as many shells use it as a command separator, so customers will have to quote the string, otherwise it will do the wrong thing, and then get an error about an unknown command being run.

Reply:
Got it~Thanks for all your clarification. 
I thought it was inconvenient to pass multiple cert fingerprints as the following:
`--certificate-fingerprint AAAA --certificate-fingerprint BBB --certificate-fingerprint CCC` ...

But then I found semicolon is used as the separator in other nuget command options as well, like `nuget restore -source`.
And dotnet restore -source addressed it with the same way here:
https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-restore#options
```
-s|--source <SOURCE>

Specifies the URI of the NuGet package source to use during the restore operation. This setting overrides all of the sources specified in the nuget.config files. Multiple sources can be provided by specifying this option multiple times. 
```
So although it's not ideal, it's inline with the convention.